From 7744504d293dc784c4de794e4eb9df25b8ff69d8 Mon Sep 17 00:00:00 2001
From: Nan Zhang <nanzhang@google.com>
Date: Thu, 8 Feb 2018 10:37:59 -0800
Subject: [PATCH 37/47] Move neon block to arm level for libvpx

After we changed the behavior for how to apply exclude_srcs to srcs,
neon srcs are missing symbols due to the lack of some arm srcs. So moved
neon block to arm level for the support, and removed the exclude_srcs.

Test: m -j checkbuild
Bug: b/70351683
Change-Id: I22e7a6ff24c8220170c3944e0815f3968b329a66
---
 Android.bp    | 12 +++---------
 Android.bp.in | 12 +++---------
 2 files changed, 6 insertions(+), 18 deletions(-)

diff --git a/Android.bp b/Android.bp
index 7f921c6..950ab3c 100644
--- a/Android.bp
+++ b/Android.bp
@@ -2082,15 +2082,9 @@ cc_library_static {
             // Want arm, not thumb, optimized
             instruction_set: "arm",
 
-            srcs: libvpx_arm_c_srcs,
-            local_include_dirs: ["config/arm"],
-
-            neon: {
-                exclude_srcs: libvpx_arm_c_srcs,
-                srcs: libvpx_arm_neon_c_srcs,
-                generated_sources: ["libvpx_arm_neon_asm_srcs_converted"],
-                local_include_dirs: ["config/arm-neon"],
-            },
+            srcs: libvpx_arm_neon_c_srcs,
+            generated_sources: ["libvpx_arm_neon_asm_srcs_converted"],
+            local_include_dirs: ["config/arm-neon"],
         },
 
         arm64: {
diff --git a/Android.bp.in b/Android.bp.in
index ac6a464..0ad23b3 100644
--- a/Android.bp.in
+++ b/Android.bp.in
@@ -24,15 +24,9 @@ cc_library_static {
             // Want arm, not thumb, optimized
             instruction_set: "arm",
 
-            srcs: libvpx_arm_c_srcs,
-            local_include_dirs: ["config/arm"],
-
-            neon: {
-                exclude_srcs: libvpx_arm_c_srcs,
-                srcs: libvpx_arm_neon_c_srcs,
-                generated_sources: ["libvpx_arm_neon_asm_srcs_converted"],
-                local_include_dirs: ["config/arm-neon"],
-            },
+            srcs: libvpx_arm_neon_c_srcs,
+            generated_sources: ["libvpx_arm_neon_asm_srcs_converted"],
+            local_include_dirs: ["config/arm-neon"],
         },
 
         arm64: {
-- 
2.17.1

